fvm/evm/stdlib/contract.go

```diff
@@ -135,6 +186,11 @@ func newInternalEVMTypeEncodeABIFunction(
 				panic(errors.NewUnreachableError())
 			}
 
+			invocation.Interpreter.ReportComputation(
+				environment.ComputationKindEVMEncodeABI,
+				calculateComputation(inter, valuesArray, evmAddressTypeID),
+			)
```

Comment:
Instead of calculating a total sum, which while it is running is itself not reported, report while "computing the computation". Basically, pass the reporting function to `calculateComputation`, and rename the function that indicates it is:
- For ABI encoding
- Reports computation (instead of computes computation)
e.g. "`reportABIEncodingComputation`"

Reply:
That's a great remark :100: , I was not aware that computation reporting does accumulate. I thought it had to be reported in one go.
This way we can abort as soon as we reach the computation limit.
Updated in https://github.com/onflow/flow-go/pull/5146/commits/32e6b6f0b533f8de085671e5fff80c4e4587118f